partner_validate_vat/res_partner.py

```diff
@@ -24,47 +24,37 @@
 #
 #
 
-from openerp.osv import osv
-from openerp.tools.translate import _
+from openerp import models, api, _
+from openerp.exceptions import ValidationError
 
 
-class ResPartner(osv.Model):
+class ResPartner(models.Model):
     _inherit = 'res.partner'
 
-    def _check_vat_uniqueness(self, cr, uid, ids, context=None):
+    @api.multi
+    @api.constrains('vat')
```

Comment:
This change have been apllied in the commit and we use a extra field compute with ```store=True```. This we allow us use sql_contraint. Thanks for you comment,has been of great use